pkg/client/selfhosted/path.go

```diff
@@ -0,0 +1,42 @@
+package selfhosted
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	regTemplate = (`(^(.*\.)?%s$)`)
+)
+
+func (c *Client) IsHost(host string) bool {
+
+	u, err := url.Parse(c.Options.URL)
+	if err != nil {
+		// If we can't parse the host given by the options, it's useless to keep running
+		log.Fatalf("failed parsing host: %s", c.Options.URL)
+		return false
+	}
+
+	urlRegex := fmt.Sprintf(regTemplate, u.Host)
+	reg := regexp.MustCompile(urlRegex)
+	return reg.MatchString(host)
+}
+
+func (c *Client) RepoImageFromPath(path string) (string, string, error) {
+	split := strings.Split(path, "/")
+
+	lenSplit := len(split)
+
+	if lenSplit >= 2 {
+		return split[lenSplit-2], split[lenSplit-1], nil
+	}
+
+	return "", "", errors.New("bad split path length")
+
```

Comment:
If `len(split) == 1` then we are using an image that has no repo. This is acceptable in the docker registry implementation at least.

`return "", split[0]`

https://hub.docker.com/_/registry

We can also remove the error.

Reply:
Should that not be `return "_", split[0]` then?